ERCS/erc-7798.md

```diff
@@ -0,0 +1,200 @@
+---
+eip: 7798
+title: Tap to Pay
+description: Initializing contactless payment transactions from EVM wallets
+author: Amhed Herrera (@amhed), Justin Lee (@JustinDLee), Arjun Dureja (@arjun-dureja)
+discussions-to: https://ethereum-magicians.org/t/erc-7798-contactless-payment/21501
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-10-25
+requires: 20, 681, 712
+---
+
+## Abstract
+This ERC defines a standard for contactless payment transactions that can allow for interoperable customer to merchant onchain payments, regardless of the wallets the customer and merchant are using.
+
+## Motivation
+Currently there is no standard mechanism in crypto to do contactless payment transactions via NFC. This ERC defines a standardized way of exchanging payment information between merchants and customers, or peer-to-peer individuals, so that efficient checkout can occur.
+
+## Specification
+
+Comprised of three parts:
+- A new Ethereum Provider JavaScript API method called `requestContactlessPayment`
+- An agreement on the payload for data exchange between the parties
+- An optional mechanism for relaying large JSON payloads using a backend relayer
+
+
+```mermaid
+sequenceDiagram
+    Sender->>PoS: Generate checkout intent
+    PoS->>NFC Relayer: Upload checkout intent JSON
+    PoS->>JSON RPC: requestContactlessPayment(uuid)
+    JSON RPC->>Wallet: Trigger NFC HCE
+    Recipient->>Wallet: Scan NFC Tag
+    Wallet->>Recipient: Checkout Intent URI
+    Recipient->>NFC Relayer: requestPaymentParams
+    NFC Relayer->>Recipient: Transaction data
+    Recipient->>Recipient: signTransaction
+    Recipient->>RPC Proxy: submitTransaction
+```
+
+
+### Use Cases
+
+#### Use Case 1: Customer purchases from a merchant
+1. A customer approaches a merchant and orders an item at their cash register (e.g. a cup of coffee).
+1. The merchant is running an app on their point-of-sale (PoS) system that allows initiating a checkout flow. 
+1. When the customer is done ordering, the merchant goes through the checkout flow up and triggers a payment request. 
+1. The merchant app calls the `requestContactlessPayment` RPC method with the expected payload. 
+1. The merchant's device emits an NFC signal via host card emulation (HCE) for the customer’s device to read.
+1. The customer, running their wallet software of choice, reads the NFC signal. The customer’s wallet constructs the necessary transaction(s) to execute the onchain checkout. 
+1. The wallet then signs and submits all transactions.
+1. Upon successful onchain execution of these transactions, the checkout is complete.
+
+#### Use Case 2: Peer-to-peer Sends
+1. Alice requests a payment on her wallet app by emitting an NFC signal. The contents of the signal is an [ERC-681](erc-681.md) URI
+> []will this only work with 681? 681 has limitations, like receiver intent which would be great to solve too.
```

Comment:
```suggestion
<!-- will this only work with 681? 681 has limitations, like receiver intent which would be great to solve too. -->
```

If you use HTML-style comments, the linter will make sure you replace them before going into Review.